product_docs/docs/biganimal/release/migration/dha_bulk_migration.mdx

```diff
@@ -0,0 +1,389 @@
+```
+
+
+* Run `pgd -f pgd-cli-config.yml show-raft` to confirm the raft leader:
+
+* Run `pgd -f pgd-cli-config.yml show-replslots` to confirm the replication slots:
+
+* Run `pgd -f pgd-cli-config.yml show-subscriptions` to confirm the subscriptions:
+
+* Run `pgd -f pgd-cli-config.yml show-groups` to confirm the groups:
+
+These commands will provide a snapshot of the state of the cluster before the migration begins.
+
+## Migrating the data
+
+This currently has to be performed in three phases.
+
+1. Transferring the “pre-data” using pg_dump and pg_restore. This exports and imports all the data definitions.
+1. Using MTK (Migration Toolkit) with the `--dataonly` option to transfer only the data from each table, repeating as necessary for each table.
+1. Transferring the “post-data” using pg_dump and pg_restore. This completes the data transfer.
+
+
+### Transferring the pre-data
+
+Use the `pg_dump` utility against the source database to dump the pre-data section in directory format.
+
+```
+pg_dump -Fd -f predata --section=pre-data -h <source-hostname> -p <source-port> -U <source-user> <source-dbname>
+```
+
+Once the pre-data has been dumped into the predata directory, it can be loaded, using `pg_restore` into the target cluster.
+
+```
+pg_restore -Fd --section=pre-data -d “host=ab-node-1-host dbname=<target-dbname> user=<target-user> options='-cbdr.ddl_locking=off -cbdr.commit_scope=local'"  predata
+```
+
+The `options=` section in the connection string to the server is important. The options disable DDL locking and sets the commit scope to `local` overriding any default commit scopes. Using `--section=pre-data` limits the restore to the configuration that precedes the data in the dump:
+
+### Transferring the data
+
+In this step, the Migration Toolkit will be used to transfer the table data between the source and target.
+
+Edit `/usr/edb/migrationtoolkit/etc/toolkit.properties`. 
+
+You will need to use sudo to raise your privilege to do this - ie. `sudo vi  /usr/edb/migrationtoolkit/etc/toolkit.properties`. 
+
+```
+SRC_DB_URL=jdbc:postgresql://<source-host>:<source-port>/<source-dbname>
+SRC_DB_USER=<source-user>
+SRC_DB_PASSWORD=<source-password>
+
+TARGET_DB_URL=jdbc:postgresql://<target-host>:<target-port>/<target-dbname>
+TARGET_DB_USER=<target-user>
+TARGET_DB_PASSWORD=<target-password>
+```
+
+Edit the relevant values into the settings:
+
+Ensure that the configuration file is owned by the user you intend to run the data transfer as and read-write only for its owner.
+
+Now, select sets of tables in the source database that should be be transferred together, ideally grouping them for redundancy in case of failure.
+
+```
+nohup /usr/edb/migrationtoolkit/bin/runMTK.sh -sourcedbtype postgres -targetdbtype postgres  -loaderCount 1  -tableLoaderLimit 1  -fetchSize 4000  -parallelLoadRowLimit 1000  -truncLoad -dataOnly -tables <tablename1>,<tablename2>,... <schemaname> > mtk.log
```

Comment:
```suggestion
nohup /usr/edb/migrationtoolkit/bin/runMTK.sh  -Dcbdr.ddl_locking=off -Dcbdr.commit_scope=local -sourcedbtype postgres -targetdbtype postgres  -loaderCount 1  -tableLoaderLimit 1  -fetchSize 4000  -parallelLoadRowLimit 1000  -truncLoad -dataOnly -tables <tablename1>,<tablename2>,... <schemaname> > mtk.log
```

Reply:
The -D option in the Migration Toolkit (MTK) is used to set system properties. It can be used to set various options that control the behavior of the migration process. The general format is -Dproperty=value.

Reply:
@sszhao1020 shouldn't we add these options to MTK as well?

Reply:
I guess ddl_locking=off is probably unnecessary, but commit_scope=local is good defense.

Reply:
For data-only migration, it's not a must 
But for -schemaonly or -index and -constraints, yes, please add this option.